### PR TITLE
fix(chat): Application attachments (Issues #2015, #2018, #2019, #2024)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

1. Attach file icon should not be shown for chat creation if selected custom application has 'Max attachments'=0
2. File extentions configured in app's settings should be displayed in Attach files modal form
3. Format of file types should be validated for field Attachment types
4. Incorrect small size of app icon on review form for publication request for application

Issues:

- Issue #2015 
- Issue #2018 
- Issue #2019 
- Issue #2024 

**UI changes**

Снимок экрана 2024-09-04 в 13 06 29
Снимок экрана 2024-09-04 в 13 08 32
Снимок экрана 2024-09-04 в 13 09 07

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2051.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/ApplicationDialog.tsx [hunk 1]
```diff
@@ -179,6 +183,17 @@ const ApplicationDialogView: React.FC<Props> = ({
     [handleDelete, setIsDeleteModalOpen],
   );
 
+  const handleAttachmentTypesError = () => {
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Common/ApplicationDialog.tsx]: lets wrap it with useCallback

File: apps/chat/src/components/Common/ApplicationDialog.tsx [hunk 2]
```diff
@@ -179,6 +183,17 @@ const ApplicationDialogView: React.FC<Props> = ({
...
+    setError('inputAttachmentTypes', {
+      type: 'manual',
+      message: t(`Please match the MIME format.`) || '',
+    });
+  };
+
+  const handleClearAttachmentTypesError = () => {
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Common/ApplicationDialog.tsx]: lets wrap it with useCallback

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2051.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2051.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2051.nightly-test.deltixhub.io
>E2E tests status: success

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**